src/sage/combinat/sf/kfpoly.py

```diff
@@ -24,22 +24,34 @@
 
 from sage.combinat.partition import _Partitions
 from sage.combinat.partitions import ZS1_iterator
+from sage.combinat.skew_partition import SkewPartitions
+from sage.combinat.skew_tableau import SemistandardSkewTableaux
 from sage.rings.polynomial.polynomial_ring import polygen
 from sage.rings.integer_ring import ZZ
 
 
 def KostkaFoulkesPolynomial(mu, nu, t=None):
     r"""
     Return the Kostka-Foulkes polynomial `K_{\mu, \nu}(t)`.
+    Here, `\mu` is a partition or a skew partition, whereas
+    `\nu` is a partition of the same size.
+
+    The Kostka-Foulkes polynomial is defined to be the sum
+    of the monomials `t^{\operatorname{charge}(T)` over all
+    semistandard tableaux `T` of shape `\lambda / \mu``,
+    where `\operatorname{charge}(T)` denotes the charge
+    of the reading word of `T`
+    (see :meth:`sage.combinat.words.finite_word.FiniteWord_class.charge`).
 
     INPUT:
 
-    - ``mu``, ``nu`` -- partitions
+    - ``mu`` -- partition or skew partition
+    - ``nu`` -- partition
     - ``t`` -- an optional parameter (default: ``None``)
 
     OUTPUT:
 
-    - the Koskta-Foulkes polynomial indexed by partitions ``mu`` and ``nu`` and
+    - the Kostka-Foulkes polynomial indexed by ``mu`` and ``nu`` and
       evaluated at the parameter ``t``.  If ``t`` is ``None`` the resulting
       polynomial is in the polynomial ring `\ZZ['t']`.
```

Comment:
```suggestion
      polynomial is in the polynomial ring `\ZZ[t]`.
```
A while-we-are-at-it fix (or this:
```suggestion
      polynomial is in the polynomial ring ``\ZZ['t']``.
```

Similar changes elsewhere in the file.